services/repository/files/content.go

```diff
@@ -203,7 +203,7 @@ func GetContents(ctx context.Context, repo *repo_model.Repository, treePath, ref
 	} else if entry.IsLink() {
 		contentsResponse.Type = string(ContentTypeLink)
 		// The target of a symlink file is the content of the file
-		targetFromContent, err := entry.Blob().GetBlobContent()
+		targetFromContent, err := entry.Blob().GetBlobContent(setting.UI.MaxDisplayFileSize)
```

Comment:
```suggestion
		targetFromContent, err := entry.Blob().GetBlobContent(1024)
```

Here it only reads a link name, I think it doesn't need to be that large. 1024 or 4096 or other either is fine, while 1024 is large enough (I do not think GItea could render a path on UI that long .....)